### PR TITLE
Signatories Carousal

### LINKED ISSUE #193: Signatories Carousel
## Description

Image

## Todo

- [ ] Implement the carousel

### PR DESCRIPTION
**Pull request by kennykos**
## Summary

Add Signatories Carousal to the Take Action Page
Closes #193 

## Screenshots

image

### CODE REVIEW

File: src/sections/take-action/signatories.tsx [hunk 1]
```diff
@@ -1,9 +1,86 @@
...
+
+  // Parse into a 2D array by numRows
+  const segmentedList: SourceData[][] = []
+  for (let i = 0; i < data.length / numRows; i += 1) {
+    segmentedList.push(data.slice(i * numRows, (i + 1) * numRows))
+  }
+
+  // Parse into an array of React nodes by numCols
+  const carouselChildren: React.ReactNode[] = []
+  const grid = `p-12 grid w-full grid-cols-${numCols}`
```
Comment (lizageorge) [inline src/sections/take-action/signatories.tsx:38]: The side padding on the Figma was greater than this, something like `p-16` might be a better fit (especially considering that we probably won't have a lot of completed fields.

Comment (lizageorge) [inline src/sections/take-action/signatories.tsx:38]: Also, was there a reason to make this a separate variable? leaving the styles in the component's className, like tailwind's default practices, might be more readable.

File: src/sections/take-action/signatories.tsx [hunk 2]
```diff
@@ -1,9 +1,86 @@
...
+  country: string | null
+  zipCode: number | null
+  bioLink: string | null
+  twitter: string | null
+  shouldEmail: boolean
+}
+
+export const parseSignatoriesToColumns = (
+  data: SourceData[] | undefined,
+  entryType: 'text' | 'html',
```
Comment (lizageorge) [inline src/sections/take-action/signatories.tsx:22]: This parameter seems to be unused

File: src/server/api/routers/signatory.ts [hunk 3]
```diff
@@ -63,7 +63,7 @@ export const signatoryRouter = createTRPCRouter({
 
   getSignatories: publicProcedure.query(async ({ ctx }) => {
     const items = await ctx.prisma.signatory.findMany({
-      orderBy: { createdAt: 'asc' },
+      orderBy: { createdAt: 'desc' },
```
Comment (lizageorge) [inline src/server/api/routers/signatory.ts:66]: Possible idea, we can discuss this more live - should we also order these results by which entries have more non-empty fields?

**Review by lizageorge — CHANGES_REQUESTED** [review]

File: src/sections/take-action/signatories.tsx (hunk 1 above)
Comment (kennykos) [reply src/sections/take-action/signatories.tsx]: Addressed in commit [abb18cf49f1f7f166f2977fd4ada6dc1e3891541](https://github.com/toriis-portal/toriis/pull/214/commits/abb18cf49f1f7f166f2977fd4ada6dc1e3891541)

Comment (kennykos) [reply src/sections/take-action/signatories.tsx]: No reason, also addressed in commit [abb18cf49f1f7f166f2977fd4ada6dc1e3891541](https://github.com/toriis-portal/toriis/pull/214/commits/abb18cf49f1f7f166f2977fd4ada6dc1e3891541)

File: src/sections/take-action/signatories.tsx (hunk 2 above)
Comment (kennykos) [reply src/sections/take-action/signatories.tsx:24]: Deleted in commit [abb18cf49f1f7f166f2977fd4ada6dc1e3891541](https://github.com/toriis-portal/toriis/pull/214/commits/abb18cf49f1f7f166f2977fd4ada6dc1e3891541)

File: src/server/api/routers/signatory.ts (hunk 3 above)
Comment (kennykos) [reply src/server/api/routers/signatory.ts:66]: Yes, let's discuss tonight!

File: src/sections/take-action/signatories.tsx [hunk 4]
```diff
@@ -1,9 +1,89 @@
...
+          .map((column, i) => (
+            <ol key={i} className="list-inside list-disc">
+              {column.map((item, j) => (
+                <li className="list-item list-outside" key={j}>
+                  <strong>{item.firstName + ' ' + item.lastName}</strong>
+                  {item.title.length > 0
+                    ? item.title[0] != ''
+                      ? ', ' + item.title.reduce((x, y) => x + '; ' + y)
+                      : ' '
+                    : ' '}
```
Comment (zorazrr) [inline src/sections/take-action/signatories.tsx:54]: Maybe make these into helper functions and reuse.

File: src/sections/take-action/signatories.tsx [hunk 5]
```diff
@@ -1,9 +1,89 @@
...
+  institution: string[]
+  email: string
+  createdAt: Date
+  city: string | null
+  country: string | null
+  zipCode: number | null
+  bioLink: string | null
+  twitter: string | null
+  shouldEmail: boolean
+}
```
Comment (zorazrr) [inline src/sections/take-action/signatories.tsx:20]: This is equivalent to the `Signatory` type imported from `@prisma/client` right?

File: src/sections/take-action/signatories.tsx [hunk 6]
```diff
@@ -1,9 +1,89 @@
...
+            </ol>
+          ))}
+      </div>,
+    )
+  }
+  return carouselChildren
+}
 
 const Signatories = () => {
+  const { data } = api.signatory.getSignatories.useQuery<Signatory[]>(
```
Comment (zorazrr) [inline src/sections/take-action/signatories.tsx:71]: Unrelated to this PR, but once we have a lot of signatories, having some pagination in this query would be good.

File: src/sections/take-action/signatories.tsx [hunk 7]
```diff
@@ -1,9 +1,89 @@
...
     <div>
-      <HighlightedTitle title="Signatories" size="large" color="clementine" />
+      <div className="flex justify-center">
+        <HighlightedTitle title="Signatories" size="large" color="clementine" />
+      </div>
+      <Carousel
+        carouselChildren={parseSignatoriesToColumns(data, 'text', 20, 1)}
+        controlSize="sm"
+        variant="lightBlue"
+      />
```
Comment (zorazrr) [inline src/sections/take-action/signatories.tsx:86]: Unrelated to this PR, just want to confirm that we don't want the footer on the take actions page?

**Review by zorazrr — CHANGES_REQUESTED** [review]
Nits.

File: src/sections/take-action/signatories.tsx (hunk 7 above)
Comment (kennykos) [reply src/sections/take-action/signatories.tsx:86]: we do, I just pushed it, great catch

File: src/sections/take-action/signatories.tsx (hunk 5 above)
Comment (zorazrr) [reply src/sections/take-action/signatories.tsx:20]: In that case we could directly use `Signatory` type rather than declaring a duplicate interface.

Comment (kennykos) [reply src/sections/take-action/signatories.tsx]: Yep, just did that, thanks for pointing that out

**Review by zorazrr — APPROVED** [review]
LGTM!

**Pull request merged**